packages/core/saved-objects/core-saved-objects-migration-server-internal/src/run_v2_migration.ts

```diff
@@ -93,13 +95,21 @@ export const runV2Migration = async (options: RunV2MigrationOpts): Promise<Migra
     defaultIndexTypesMap: options.defaultIndexTypesMap,
   });
 
-  // we create 2 synchronization objects (2 synchronization points) for each of the
+  // we create synchronization objects (synchronization points) for each of the
   // migrators involved in relocations, aka each of the migrators that will:
   // A) reindex some documents TO other indices
   // B) receive some documents FROM other indices
   // C) both
-  const readyToReindexDefers = createMultiPromiseDefer(indicesWithMovingTypes);
-  const doneReindexingDefers = createMultiPromiseDefer(indicesWithMovingTypes);
+  const readyToReindexWaitGroupMap = createWaitGroupMap(indicesWithMovingTypes);
+  const doneReindexingWaitGroupMap = createWaitGroupMap(indicesWithMovingTypes);
+  const updateAliasesWaitGroupMap = createWaitGroupMap<AliasAction[], Promise<any>>(
+    indicesWithMovingTypes,
+    (allAliasActions) =>
+      updateAliases({
+        client: options.elasticsearchClient,
+        aliasActions: allAliasActions.flat(),
+      })()
```

Comment:
Yes, that's correct!

Reply:
This is one of the simplest solutions I came up with, but perhaps we can create a separate issue to refactor and make something more centralised, some sort of `SynchronizationManager`.

Reply:
I had the same "Oh!" moment 😅 I would not come here to look for an `updateAliases` action call. I think it's fine to merge as-is and rather spend extra time on testing. 

But I think it would be worth exploring if we could make this less "surprising". One way might be to let _every_ migrator call the updateAliases action. Before we had 6 update aliases calls with each call doing one index/alias. Now we'd have 1 batch call and 5 no-ops.

Another option could be to use the `synchronizeMigrators.then` hook but only let the `.kibana` migrator do the updateAliases call, other migrators have a no-op then hook.